# LCA caluculation: a simple example

[source](https://link.springer.com/book/10.1007/978-94-015-9900-9) of this example

For instance, consider a unit process (or process in short), say, **production of electricity**, which: 
* uses 2 litre of fuel
* produces 10 kWh of electricity
* emits 1 kg of carbon dioxide 
* emits 0.1 kg of sulphur

Then a second unit process, say **production of fuel** that
* produces 100 litre of fuel
* emits 10 kg of carbon dioxide
* emits 2 kg of sulphur dioxide
* requires 50 litre of crude oil

The process matrix can be represented as 

![](process_matrix.png)


**Matrix A** is the **technology matrix**; it represents the flows within the economic systems. 

**Matrix B** is the **intervention matrix**; it represents the environmental interventions of unit processes. 

Let's represent A and B with Numpy:

In [2]:
import numpy as np

A = np.array([[-2, 100],[10, 0]])
B = np.array([[1, 10],[0.1, 2],[0, -50]])
A, B

(array([[ -2, 100],
        [ 10,   0]]),
 array([[  1. ,  10. ],
        [  0.1,   2. ],
        [  0. , -50. ]]))

Next, let's define a **functional unit**, in this example the production of 1000 kWh of electricity

In [3]:
f = np.array([0, 1000]) # reference flow

We now introduce the **scaling vector s**, which tells us how much of process 1 and 2 we need to produce our functional unit

![](equations.png)

We can solve this linear system using

![](inverse.png)

In Numpy, that gives

In [4]:
s = np.matmul(np.linalg.inv(A), f)
s

array([100.,   2.])

In our solution, we will need:

* 100 units of process 1 (production of electricity) and 
* 2 units of process 2 (production of fuel) 

to produce 1 functional unit

# Environmental flows

We can now use **s** to compute the environmental flows, using **g=Bs**

In [16]:
g = np.matmul(B, s)
g

array([ 120.,   14., -100.])

Thus the environmental flows will be:
* 120 kg of carbon dioxide emitted
* 14 kg of sulphur dioxide emitted
* 100 litre of crude oil consumed


